server/routes/slack/slack_routes.py

```
@@ -162,13 +170,19 @@ def slack_callback():
                 "incidents_channel_name": channel_result.get('channel_name'),
             }
             store_tokens_in_db(user_id, slack_token_data, "slack")
+            
+            # Persist channel ID as org preference so it survives disconnect/reconnect
+            if org_id and channel_result.get('channel_id'):
+                store_org_preference(org_id, 'slack_incidents_channel_id', channel_result['channel_id'])
+                store_org_preference(org_id, 'slack_incidents_channel_name', channel_result.get('channel_name', ''))
+            
             logging.info("Incidents channel ready, Slack credentials stored successfully")
         except Exception as e:
             logging.error("Failed to store Slack credentials", exc_info=True)
```

Ruff (0.15.17)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)